### ISSUE TITLE
Crash when decoding JPEG image

### ISSUE THREAD

**Issue by samueltardieu** [cgeo/cgeo#6261]
Reported on support for cgeo 2017-01-12-RC1.

> Bug: if I open up GC15NP2 (not saved), swipe right to the finders tab, and then right again, I get a crash. Please let me know if you want a debug log.

```
--- System information ---
Device: SM-N900P (cm_hltespr, samsung)
Android version: 6.0.1
Android build: MOB31K test-keys
Cgeo version: 2017.01.12-RC1
Google Play services: enabled - 10.0.84 (438-137749526)
Low power mode: inactive
Compass capabilities: yes
Rotation vector sensor: present
Orientation sensor: present
Magnetometer & Accelerometer sensor: present
Direction sensor used: rotation vector
Hide own/found: false
Map strategy: auto
HW acceleration: enabled (default state)
System language: en_US
Log date format: 1/15/17
Debug mode active: no
Fine location permission: granted
Write external storage permission: granted
Geocaching sites enabled:
   geocaching.com: Logged in (Login OK) / PREMIUM
Geocaching.com date format: MM/dd/yyyy
Installed cgeo plugins:  none
--- End of system information ---
```

**Comment by samueltardieu** [comment]
I could reproduce this:

```01-16 17:41:51.901 D/cgeo    ( 9832): [OkHttp] GET https://img.geocaching.com/cache/large/8adefbb1-fe63-443f-a99b-61385b44ea04.jpg
01-16 17:41:51.923 D/cgeo    ( 9832): [OkHttp] GET https://imgcdn.geocaching.com/cache/log/large/dea1f733-6402-4307-b5a5-057d5b57fc0b.jpg
01-16 17:41:51.934 D/cgeo    ( 9832): [OkHttp] GET https://imgcdn.geocaching.com/cache/log/large/e92c223b-5d6f-4133-b5e7-7ffd013386a3.jpg
01-16 17:41:51.940 D/cgeo    ( 9832): [OkHttp] GET https://imgcdn.geocaching.com/cache/log/large/7e9426aa-c315-4969-b9af-83f68e008641.jpg
01-16 17:41:51.951 D/cgeo    ( 9832): [OkHttp] GET https://imgcdn.geocaching.com/cache/log/large/04cb208d-ea9f-493c-8636-d03c0cc43a0b.jpg
01-16 17:41:51.971 D/cgeo    ( 9832): [OkHttp] GET https://imgcdn.geocaching.com/cache/log/large/8fc4c69e-bb64-44a0-bb21-4dd0d935e592.jpg
01-16 17:41:52.058 W/cgeo    ( 9832): [main] UncaughtException
01-16 17:41:52.058 W/cgeo    ( 9832): java.lang.RuntimeException: Undefined Prefix: geocache in org.kxml2.io.KXmlParser@8fdf06b
01-16 17:41:52.058 W/cgeo    ( 9832): 	at org.kxml2.io.KXmlParser.adjustNsp(KXmlParser.java:260)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at org.kxml2.io.KXmlParser.parseStartTag(KXmlParser.java:1153)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at org.kxml2.io.KXmlParser.next(KXmlParser.java:372)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at org.kxml2.io.KXmlParser.nextToken(KXmlParser.java:317)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at org.apache.harmony.xml.parsers.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:328)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at org.apache.harmony.xml.parsers.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:295)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at org.apache.harmony.xml.parsers.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:295)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at org.apache.harmony.xml.parsers.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:128)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.adobe.xmp.impl.XMPMetaParser.parseInputSource(XMPMetaParser.java:273)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.adobe.xmp.impl.XMPMetaParser.parseXmlFromBytebuffer(XMPMetaParser.java:189)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.adobe.xmp.impl.XMPMetaParser.parseXml(XMPMetaParser.java:132)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.adobe.xmp.impl.XMPMetaParser.parse(XMPMetaParser.java:77)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.adobe.xmp.XMPMetaFactory.parseFromBuffer(XMPMetaFactory.java:160)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.adobe.xmp.XMPMetaFactory.parseFromBuffer(XMPMetaFactory.java:144)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.drew.metadata.xmp.XmpReader.extract(XmpReader.java:130)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.drew.metadata.xmp.XmpReader.extract(XmpReader.java:114)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.drew.metadata.xmp.XmpReader.readJpegSegments(XmpReader.java:103)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.drew.imaging.jpeg.JpegMetadataReader.processJpegSegmentData(JpegMetadataReader.java:130)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.drew.imaging.jpeg.JpegMetadataReader.process(JpegMetadataReader.java:122)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.drew.imaging.jpeg.JpegMetadataReader.readMetadata(JpegMetadataReader.java:73)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.drew.imaging.jpeg.JpegMetadataReader.readMetadata(JpegMetadataReader.java:80)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.drew.imaging.ImageMetadataReader.readMetadata(ImageMetadataReader.java:112)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.drew.imaging.ImageMetadataReader.readMetadata(ImageMetadataReader.java:162)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at cgeo.geocaching.ui.ImagesList.getImageLocation(ImagesList.java:195)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at cgeo.geocaching.ui.ImagesList.display(ImagesList.java:181)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at cgeo.geocaching.ui.ImagesList.access$100(ImagesList.java:61)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at cgeo.geocaching.ui.ImagesList$2.accept(ImagesList.java:147)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at cgeo.geocaching.ui.ImagesList$2.accept(ImagesList.java:144)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:59)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at io.reactivex.internal.operators.observable.ObservableTakeWhile$TakeWhileObserver.onNext(ObservableTakeWhile.java:90)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:198)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:250)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:109)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at android.os.Handler.handleCallback(Handler.java:751)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at android.os.Handler.dispatchMessage(Handler.java:95)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at android.os.Looper.loop(Looper.java:154)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at android.app.ActivityThread.main(ActivityThread.java:6119)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at java.lang.reflect.Method.invoke(Native Method)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
01-16 17:41:52.058 W/cgeo    ( 9832): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
01-16 17:41:52.059 E/AndroidRuntime( 9832): FATAL EXCEPTION: main
01-16 17:41:52.059 E/AndroidRuntime( 9832): Process: cgeo.geocaching, PID: 9832
```

**Comment by Lineflyer** [comment]
Does this problem justify to stop the staged release and create a new release, or is it a rare case?

**Comment by samueltardieu** [comment]
I think this is a rare case of a malformed JPEG image.

**Comment by samueltardieu** [comment]
But if those tasks should not be added to the "Next bugfix release" project, please tell me, I understand this might create some confusion if this is only about the already existing release.

**Comment by B2Pi** [comment]
Confirming that the issue is fixed in the latest RC (2017.02.05-RC) Cheers
